### PR TITLE
[Extra/rest 2.5] support new obi result

### PR DESCRIPTION
**Pull request by satawatnack**
## Implementation details
- Implement decode result to support new obi result format
- add test

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: hooks/price/types.go [hunk 1]
```diff
@@ -1,10 +1,98 @@
...
+type Input struct {
+	Symbols            []string
+	MinimumSourceCount uint8
+}
+
+type Output struct {
+	Responses []Responses
+}
+
+type Responses struct {
```
Comment (RogerKSI) [inline hooks/price/types.go:33]: should be Response? why plural

File: hooks/price/types.go [hunk 2]
```diff
@@ -1,10 +1,98 @@
...
+	Symbol       string
+	ResponseCode uint8
+	Rate         uint64
+}
+
+func MustDecodeResult(calldata, result []byte) CommonOutput {
+	var symbols []string
+	var rates []uint64
+
+	responses, err := DecodeResult(result)
```
Comment (RogerKSI) [inline hooks/price/types.go:43]: should DecodeResult and DecodeLegacyResult return CommonOutput directly? like we decode it to our common standard. 

think it's cleaner as this function will only just try each standard.

**Review by RogerKSI — CHANGES_REQUESTED** [review]

File: hooks/price/types.go [hunk 3]
```diff
@@ -1,10 +1,98 @@
...
+}
+
+type Output struct {
+	Responses []Responses
+}
+
+type Responses struct {
+	Symbol       string
+	ResponseCode uint8
+	Rate         uint64
```
Comment (colmazia) [inline hooks/price/types.go:36]: why new OS not return multiplier na? @warittornc

File: hooks/price/types.go (hunk 2 above)
Comment (colmazia) [reply hooks/price/types.go:43]: I prefer this current way na

File: hooks/price/types.go (hunk 3 above)
Comment (warittornc) [reply hooks/price/types.go:36]: the multiplier is hardcoded into the oracle script, returning it is unnecessary

File: hooks/price/types.go [hunk 4]
```diff
@@ -1,10 +1,98 @@
 package price
 
-type Input struct {
+import (
+	"github.com/bandprotocol/chain/v2/pkg/obi"
+)
+
+const DefaultMultiplier = uint64(100000000)
```
Comment (warittornc) [inline hooks/price/types.go:7]: 10^9

File: hooks/price/types.go (hunk 2 above)
Comment (warittornc) [reply hooks/price/types.go:43]: I prefer Decode_() return a common result and return that directly

**Review by warittornc — APPROVED** [review]

**Review by RogerKSI — APPROVED** [review]

**Review by taobun — APPROVED** [review]

**Comment by RogerKSI** [comment]
Note: merge this after upgrade v2.5

**Pull request merged**